test-specifications/utility.md

```diff
@@ -0,0 +1,293 @@
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 763543,
+  "method": "setup",
+  "params": {
+    "operatorAccountId": "0.0.47762334",
+    "operatorPrivateKey": "302e020100300506032b65700422042091f37373fe8b38bd4495e489ae7cb50c28909970231b906b6322a984e582f6af",
+    "nodeIp": "127.0.0.1:50211",
+    "nodeAccountId": "0.0.3",
+    "mirrorNetworkIp": "127.0.0.1:5600"
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 763543,
+  "result": {
+    "message": "Successfully setup custom client.",
+    "status": "SUCCESS"
+  }
+}
+```
+
+---
+
+### `reset`
+
+#### Description
+
+Method used to close the TCK network connections. Network connections can be reestablished after with another `setup` call.
+
+#### Parameters
+
+| Parameter Name | Type   | Required/Optional | Input/Output | Description/Notes                                       |
+|----------------|--------|-------------------|--------------|---------------------------------------------------------|
+| message        | string | required          | Output       | Informational message about the execution of the method |
+| status         | string | required          | Output       | The status/result of the execution                      |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "method": "reset"
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "result": {
+    "message": "Successfully reset client.",
+    "status": "SUCCESS"
+  }
+}
+```
+
+---
+
+### `generateKey`
+
+#### Description
+
+Method used to generate a Hedera Key.
+
+#### Parameters
+
+| Parameter Name | Type   | Required/Optional | Input/Output | Description/Notes                                                                                                                                                                                                                                                                                                                             |
+|----------------|--------|-------------------|--------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| type           | string | optional          | Input        | The type of Key to generate. If provided, it MUST be one of `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, `ecdsaSecp256k1PublicKey`, `keyList`, or `thresholdKey`. If not provided, the returned key will be of type `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, or `ecdsaSecp256k1PublicKey` |
```

Comment:
> If not provided, the returned key will be of type `ed25519PrivateKey`, `ed25519PublicKey`, `ecdsaSecp256k1PrivateKey`, or `ecdsaSecp256k1PublicKey`

Why do we need random key type output? Is there a test case for this?

Reply:
I thought this would mostly for developer convenience, if a key is needed and the developer doesn't really care what type it is. Otherwise for every test where a key is required we'll have to specify what kind of key